src/libraries/Microsoft.PowerFx.Core/Public/DisplayNames/RenameDriver.cs

```diff
@@ -61,23 +66,97 @@ private static FormulaType RenameFormulaTypeHelper(AggregateType nestedType, Que
                 };
                 var newProvider = new SingleSourceDisplayNameProvider(names);
 
-                return FormulaType.Build(DType.ReplaceDisplayNameProvider(DType.DisableDisplayNameProviders(nestedType._type), newProvider));
+                return GetWrappedAggregateType(nestedType, newProvider);
             }
 
             if (!nestedType.TryGetFieldType(field, out var fieldType) || fieldType is not AggregateType aggregateType)
             {
                 // Path doesn't exist within parameters, return as is, stripping displaynameproviders
-                return FormulaType.Build(DType.DisableDisplayNameProviders(nestedType._type));
+                return GetWrappedAggregateType(nestedType, DisabledDisplayNameProvider.Instance);
             }
 
             var innerUpdatedType = RenameFormulaTypeHelper(aggregateType, segments, updatedName);
-            var fError = false;
 
-            // Use DType internals to swap one field type for the updated one and disable all other display names
-            var dropped = DType.DisableDisplayNameProviders(nestedType._type.Drop(ref fError, DPath.Root, field));
-            Contracts.Assert(!fError);
+            // Wrap the nestedType, swapping one field for the updated one and disabling all other display names
+            return GetWrappedAggregateType(nestedType, DisabledDisplayNameProvider.Instance, field, innerUpdatedType);
+        }
+
+        private AggregateType GetWrappedAggregateType(AggregateType rootType, DisplayNameProvider displayNameProvider, string replacedFieldName = null, FormulaType replacedFieldType = null)
+        {
+            if (!rootType._type.IsLazyType)
```

Comment:
>_type [](http://example.com/codeflow?start=26&length=5)

Is "_" used even for internal type in naming? I thought that this was only used for private members.

Reply:
No, this should be changed at some point